### ISSUE TITLE
Asia language support

### ISSUE THREAD

**Issue by Eudemon** [spencermountain/wtf_wikipedia#424]
for example parsing https://ja.wikipedia.org/wiki/%E3%81%82%E3%82%8A%E3%81%B5%E3%82%8C%E3%81%9F%E8%81%B7%E6%A5%AD%E3%81%A7%E4%B8%96%E7%95%8C%E6%9C%80%E5%BC%B7 to get released volume information

via `wtf.fetch("ありふれた職業で世界最強", "ja")`, `doc.infoboxes(1)` show data as [object object]

tried i18n and html plugin, which returns `<td>[object Object]</td>`

**Comment by spencermountain** [comment]
hi Randy! thanks for the issue.
You're right. There is a bug here. It's not parsing the infobox properties with japanese characters.
```js
let str = `{{Infobox animanga/Novel
| 著者 = [[白米良]]
| イラスト = [[たかやKi]]
| f出版社 = [[オーバーラップ (企業)|オーバーラップ]]
| 掲載誌 = [[小説家になろう]]
| レーベル = [[オーバーラップ文庫]]
| 開始号 = 2013年11月7日
| 終了号 = 
| 開始日 = 2015年6月24日
| 終了日 = 
| 巻数 = 既刊16巻（本編11巻+外伝4巻+短編集1巻）<br />（2020年7月現在）
| インターネット = 1
}}`
let doc = wtf(str)
console.log(doc.infoboxes(0).json())
```
it's also really weird how we're getting stringified `[object Object]` in there too - I've never seen that before.

I can try releasing a fix this week, otherwise it may fall into the v9 release, which may take longer
ありがとうございました！

**Comment by spencermountain** [comment]
hey Randy, I found a fix, but think it's best to include this in v9, which may be a few weeks away still. 

if you want a fix now, you can change this line in [/src/template/_parsers/02-keyMaker.js](https://github.com/spencermountain/wtf_wikipedia/blob/master/src/template/_parsers/02-keyMaker.js#L3)
to support japanese unicode characters:
`const hasKey = /^[a-z0-9\u00C0-\u00FF\u4e00-\u9faf\._\- '()œ]+=/iu`
to fix this issue.

This is code that has changed a lot on the dev branch, for our next big release, so I'm scared of doing a hotfix for this here.

we really need a better top-to-bottom solution for unicode regular expressions! But this should work for now.
cheers

**Comment by Eudemon** [comment]
> 
> 
> hey Randy, I found a fix, but think it's best to include this in v9, which may be a few weeks away still.
> 
> if you want a fix now, you can change this line in [/src/template/_parsers/02-keyMaker.js](https://github.com/spencermountain/wtf_wikipedia/blob/master/src/template/_parsers/02-keyMaker.js#L3)
> to support japanese unicode characters:
> `const hasKey = /^[a-z0-9\u00C0-\u00FF\u4e00-\u9faf\._\- '()œ]+=/iu`
> to fix this issue.
> 
> This is code that has changed a lot on the dev branch, for our next big release, so I'm scared of doing a hotfix for this here.
> 
> we really need a better top-to-bottom solution for unicode regular expressions! But this should work for now.
> cheers

Hi, I tried manually editing the file from node_modules to what you suggested, and ran the test again still got same [object object] response

also note that this happens to not just Japanese characters, same result with Chinese etc

thanks for the response こちらこそありがとうございます

**Comment by spencermountain** [comment]
oh yeah, if you are editing the node_module files directly, you may need to make sure you are not using the build, so your import will have to look something like `../node_modles/wtf_wikipedia/src` - depening on your setup.

You're right - we're gonna need to start using unicode regex properties. They are now [pretty-well supported](https://caniuse.com/?search=unicode-property-escapes)
cheers

**Comment by Eudemon** [comment]
oh yea, totally forgot about /build, knew I was doing something wrong
it works now! thank you for the quick response, looking forward to v9!

**Comment by Eudemon** [comment]
update: items within the initial data array still shows [object object], seems the change doesn't apply to nested array

![Screenshot 2021-03-22 125610](https://user-images.githubusercontent.com/330995/112028112-37fc0b80-8b0e-11eb-850a-364c1dcb5e31.jpg)

**Comment by Eudemon** [comment]
sorry for bump, just wanted to make sure you seen my last message since i responded fairly quick with previous

**Comment by spencermountain** [comment]
yeah, a fix is on dev, release is behind a bunch of other things now.
thanks for your patience

**Comment by spencermountain** [comment]
this is fixed now on 9.0.0
thanks for your help
